extensions/applicationinsights-clickanalytics-js/src/events/WebEvent.ts

```diff
@@ -0,0 +1,153 @@
+        this._setBasicProperties(event, overrideValues);
+        this._setPageTags(event, overrideValues);
+
+        // extract specific meta tags out of the pageTags.metaTags collection.  These will go into assigned first class fields in the event.
+        // the rest will go into pageTags.metaTags collection as is.
+        this._pageTypeMetaTag = this._getMetaData(this._metaTags, this._config.coreData, 'pageType');
+        this._behaviorMetaTag = this._getMetaData(this._metaTags, this._config.coreData, 'behavior');
+
+        if (isValueAssigned(overrideValues.pageType)) {
+            event.pageType = overrideValues.pageType;
+        }
+        // Only assign if not overriden and meta data is available
+        if (isValueAssigned(this._pageTypeMetaTag) && !isValueAssigned(event.pageType)) {
+            event.pageType = this._pageTypeMetaTag;
+        }
+        if (isValueAssigned(this._marketMetaTag)) {
+            event.market = this._marketMetaTag;
+        }
+        
+        
+    }
+
+    /**
+     * Sets pageTags.
+     * @param event - The event
+     */
+    protected _setPageTags(event: ITelemetryEventInternal, overrideValues: IOverrideValues) {
+        // Prepare the pageTags object that is mostly the same for all events.  Event specific pageTags will be added inside event constructors.
+       
+        if (this._pageTagsCallback) {
+            this._pageTags = extend(true, this._pageTags, this._pageTagsCallback());
+        }
+        if (isValueAssigned(overrideValues.pageTags)) {
+            this._pageTags = extend(true, this._pageTags, overrideValues.pageTags);
+        }
+        // If metadata is present add it to pageTags property
+        if (this._metaTags) {
+            this._pageTags.metaTags = {};
+            // Remove not supported meta data in pageTags.metaTags
+            for (var metaTag in this._metaTags) {
+                if (metaTag != 'behavior' && metaTag != 'market' && metaTag != 'pageType') {
+                    this._pageTags.metaTags[metaTag] = this._metaTags[metaTag];
+                }
+            }
+        }
+        // All metadata tags that must be saved as properties have been extracted at this point.  Assign pageTags as is.
+        event.properties = event.properties || {};
+        event.properties['pageTags'] = this._pageTags;
+    }
+
+    protected _getBehavior(overrideValues?: IOverrideValues): number {
+        let behavior: any;
+        // If override specified 
+        if (overrideValues && isValueAssigned(overrideValues.behavior)) {
+            behavior = overrideValues.behavior;
+        }
+        // If behavior meta tag available
+        else if (isValueAssigned(this._behaviorMetaTag)) {
+            behavior = this._behaviorMetaTag;
+        }
+        return this._getValidBehavior(behavior);
+    }
+
+    protected _getValidBehavior(behavior: any): number {
+        if (isValueAssigned(behavior)) {
```

Comment:
Are we going to support strings here or just numbers?

Reply:
Need to check with Heart Analytics team

Reply:
We will be supporting Strings and Numbers as per the new implementation.